simpleline/event_loop/main_loop.py

```diff
@@ -110,10 +123,22 @@ def enqueue_signal(self, signal):
         Enqueue signal to the most inner queue (nearest to the active queue) where the `signal.source` belongs.
         If it belongs nowhere enqueue it to the active one.
 
+        This method is thread safe.
+
         :param signal: event which you want to add to the event queue for processing
         :type signal: instance based on AbstractEvent class
         """
-        for queue in reversed(self._event_queues):
+        # TODO: Remove when python3-astroid 1.5.3 will be in Fedora
+        # pylint: disable=not-context-manager
+        with self._lock:
+            count = len(self._event_queues)
```

Comment:
So true, I've missed that.
I'm trying to be defensive when locks get into play. However, right here it doesn't give sense I guess. If I didn't missed anything it should be fine when the whole `for` part will be under lock. It will be double lock because `EventQueue` `source` lock, however, there is no higher logic and I think there is no way how the dead lock/starvation could arise here.